systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java

```diff
@@ -745,6 +751,138 @@ void testMultiNodeKafkaConnectWithConnectorCreation() {
+                    .addToConfig("config.storage.topic", connectClusterName + "-config")
+                    .addToConfig("status.storage.topic", connectClusterName + "-status")
+                    .withNewTls()
+                        .withTrustedCertificates(certSecret)
+                    .endTls()
+                    .withBootstrapServers(CLUSTER_NAME + "-kafka-bootstrap:9093")
+                .endSpec()
+                .done();
+
+        testConnectAuthorizationWithWeirdUserName(weirdUserName, SecurityProtocol.SSL);
+    }
+
+    @Tag(NODEPORT_SUPPORTED)
+    @Tag(EXTERNAL_CLIENTS_USED)
+    @Tag(CONNECTOR_OPERATOR)
+    @Test
+    void testConnectScramShaAuthWithWeirdUserName() {
+        // Create weird named user with . and more than 64 chars -> SCRAM-SHA
+        String weirdUserName = "jjglmahyijoambryleyxjjglmahy.ijoambryleyxjjglmahyijoambryleyxasdsadasdasdasdasdgasgadfasdad";
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3)
+                .editSpec()
+                    .editKafka()
+                        .withNewListeners()
+                            .withNewTls()
+                                .withAuth(new KafkaListenerAuthenticationScramSha512())
+                            .endTls()
+                            .withNewKafkaListenerExternalNodePort()
+                                .withAuth(new KafkaListenerAuthenticationScramSha512())
+                            .endKafkaListenerExternalNodePort()
+                        .endListeners()
+                    .endKafka()
+                .endSpec()
+                .done();
+
+        KafkaUserResource.scramShaUser(CLUSTER_NAME, weirdUserName).done();
+
+        KafkaConnectResource.kafkaConnect(CLUSTER_NAME, 1)
+                .editMetadata()
+                    .addToLabels("type", "kafka-connect")
+                    .addToAnnotations(Annotations.STRIMZI_IO_USE_CONNECTOR_RESOURCES, "true")
+                .endMetadata()
+                .withNewSpec()
+                    .withBootstrapServers(KafkaResources.tlsBootstrapAddress(CLUSTER_NAME))
+                    .withNewKafkaClientAuthenticationScramSha512()
+                        .withNewUsername(weirdUserName)
+                        .withPasswordSecret(new PasswordSecretSourceBuilder()
+                            .withSecretName(weirdUserName)
+                            .withPassword("password")
+                            .build())
+                    .endKafkaClientAuthenticationScramSha512()
+                    .addToConfig("key.converter.schemas.enable", false)
+                    .addToConfig("value.converter.schemas.enable", false)
+                    .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                    .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+                    .withVersion(Environment.ST_KAFKA_VERSION)
+                    .withReplicas(1)
```

Comment:
This property is redundant, you have already define it in the `KafkaConnectResource.kafkaConnect(CLUSTER_NAME, 1)`